mlflow/server/js/src/reducers/Reducers.js

```diff
@@ -49,21 +57,24 @@ export const getRunInfo = (runUuid, state) => {
 
 const runInfosByUuid = (state = {}, action) => {
   switch (action.type) {
-    case fulfilled(GET_EXPERIMENT_API): {
-      let newState = { ...state };
+    case fulfilled(GET_RUN_API): {
+      const runInfo = RunInfo.fromJs(action.payload.run.info);
+      return amendRunInfosByUuid(state, runInfo);
+    }
+    case fulfilled(SEARCH_RUNS_API): {
+      const newState = {};
```

Comment:
`runInfosByUuid` will now be the source of truth for runs to display now. Old runs will be thrown away after we call search.